edocumentation-service/src/main/java/org/dcsa/edocumentation/service/unofficial/UnofficialTransportDocumentService.java

```diff
@@ -64,6 +63,22 @@ public Optional<TransportDocumentRefStatusTO> issueDraft(DraftTransportDocumentR
     if (carrier == null) {
       throw ConcreteRequestErrorMessageException.notFound("Unknown carrier SMDG code: " + carrierSMDGCode);
     }
+
+    boolean shouldHaveDeclaredValue = shippingInstruction.getConsignmentItems()
+      .stream()
+      .map(ConsignmentItem::getShipment)
+      .map(Shipment::getBooking)
+      .map(Booking::getDeclaredValue)
+      .anyMatch(Objects::nonNull);
+
+    if (shouldHaveDeclaredValue ^ transportDocumentRequestTO.declaredValue() != null) {
```

Comment:
How do we test this condition? There are no new unit tests or postman tests to cover edge cases for this logic

Reply:
Existing if condition covers 2 use cases:
1. oneOrMoreSiBookingsHaveDeclaredValue == false && transportDocumentRequestTO.declaredValue() != null
2. oneOrMoreSiBookingsHaveDeclaredValue == true && transportDocumentRequestTO.declaredValue() == null

Existing error message covers only 2st condition above. Please consider breaking down if condition into 2 separate conditions with separate error messages to capture both situations, or rewriting if statement to cover only 2nd condition if that was the intended logic

Reply:
Fixed the error message to distinguish.  For the test case, I agree but I would like to punt it to a later PR for this issue (to minimize the changes to the postman test)